aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StoragePathGetPresignedUrlOperation.kt

```diff
@@ -48,6 +49,30 @@ internal class AWSS3StoragePathGetPresignedUrlOperation(
                 return@submit
             }
 
+            if (request.validateObjectExistence) {
+                try {
+                    storageService.validateObjectExists(serviceKey)
+                } catch (nfe: NotFound) {
```

Comment:
This seems odd, `storageService` is of type **StorageService**, yet we're catching a specific Kotlin SDK exception. If we need to swap out the implementation or mock the returned error, we'll be dependent on the Kotlin SDK.

Reply:
I've refactored the exception handling to isolate `aws.sdk.kotlin.services.s3.model.NotFound` to `StorageService`.